packages/eslint-plugin/src/rules/no-unnecessary-qualifier.ts

```diff
@@ -0,0 +1,208 @@
+      const fromScope = getSymbolInScope(
+        tsQualifier,
+        accessedSymbol.flags,
+        sourceCode.getText(name)
+      );
+
+      return (
+        typeof fromScope === 'undefined' ||
+        symbolsAreEqual(accessedSymbol, fromScope)
+      );
+    }
+
+    function visitNamespaceAccess(
+      node: TSESTree.Node,
+      qualifier: TSESTree.Node,
+      name: TSESTree.Identifier
+    ): void {
+      // Only look for nested qualifier errors if we didn't already fail on the outer qualifier.
+      if (
+        !currentFailedNamespaceExpression &&
+        qualifierIsUnnecessary(qualifier, name)
+      ) {
+        currentFailedNamespaceExpression = node;
+        context.report({
+          node: qualifier,
+          messageId: 'unnecessaryQualifier',
+          data: {
+            name: sourceCode.getText(name)
+          },
+          fix(fixer) {
+            return fixer.removeRange([qualifier.range[0], name.range[0]]);
+          }
+        });
+      }
+    }
+
+    function enterDeclaration(node: TSESTree.Node): void {
+      const tsDeclaration = esTreeNodeToTSNodeMap.get(node);
+      if (tsDeclaration) {
+        namespacesInScope.push(tsDeclaration);
+      }
+    }
+
+    function exitDeclaration(node: TSESTree.Node) {
+      if (esTreeNodeToTSNodeMap.has(node)) {
+        namespacesInScope.pop();
+      }
+    }
+
+    function resetCurrentNamespaceExpression(node: TSESTree.Node): void {
+      if (node === currentFailedNamespaceExpression) {
+        currentFailedNamespaceExpression = null;
+      }
+    }
+
+    function isPropertyAccessExpression(
+      node: TSESTree.Node
+    ): node is TSESTree.MemberExpression {
+      return node.type === 'MemberExpression' && !node.computed;
+    }
+
+    function isEntityNameExpression(node: TSESTree.Node): boolean {
+      return (
+        node.type === 'Identifier' ||
+        (isPropertyAccessExpression(node) &&
+          isEntityNameExpression(node.object))
+      );
+    }
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      TSModuleDeclaration: enterDeclaration,
+      TSEnumDeclaration: enterDeclaration,
+      'ExportNamedDeclaration[declaration.type="TSModuleDeclaration"]': enterDeclaration,
+      'ExportNamedDeclaration[declaration.type="TSEnumDeclaration"]': enterDeclaration,
+      'TSModuleDeclaration:exit': exitDeclaration,
+      'TSEnumDeclaration:exit': exitDeclaration,
+      'ExportNamedDeclaration[declaration.type="TSModuleDeclaration"]:exit': exitDeclaration,
+      'ExportNamedDeclaration[declaration.type="TSEnumDeclaration"]:exit': exitDeclaration,
+      TSQualifiedName(node: TSESTree.TSQualifiedName): void {
+        visitNamespaceAccess(node, node.left, node.right);
+      },
+      MemberExpression(node: TSESTree.MemberExpression): void {
+        if (node.computed) return;
```

Comment:
You could handle this in the selector instead:
`'MemberExpression[computed=false]'`

Reply:
Fixed. Still getting used to that selector syntax :)

Reply:
so am I!
it's so powerful but it's so hard to think about it.
I've found it much easier to pick out when reviewing code.